test/integration/backward_compatible/parallel/map/NearCachedMapTest.js

```diff
@@ -96,15 +96,17 @@ describe('NearCachedMapTest', function () {
                 expectStats(map1, 0, 2, 1);
             });
 
-            it('get returns null if the entry was removed by another client', async function () {
+            it('get returns null eventually after the entry is removed by another client', async function () {
                 if (!invalidateOnChange) {
                     this.skip();
                 }
                 await map1.get('key1');
                 await map2.remove('key1');
-                const val = await TestUtil.promiseLater(1000, map1.get.bind(map1, 'key1'));
-                expectStats(map1, 0, 2, 1);
-                expect(val).to.be.null;
+                const val = await map1.get('key1');
```

Comment:
Can't this result in a hit and return non-null from the client's near-cache, if the invalidation is not yet received by the client?

Reply:
🤔 yes it can, my code feels wrong. The get call was in assertTrueEventually but I don't know if it will have side affects that will make test fail.